### ISSUE TITLE
[P2] Tensor-parallel shard reparse replaces the parent's encoder fixture identity

### ISSUE THREAD

**Issue by RobTand** [RobTand/tessera#236]
Audited source: `RobTand/tessera` at `ea72d23cbf26a453d0d740afac537fb6d6b8afbc` (2026-09-05).

**Trigger and impact.** A TP loader cuts an artifact produced by a different/older encoder fixture, then reparses that shard. `_reparse_shard` calls `build_unit_artifact` without `fixture_id`, so the new manifest records the current encoder fixture even though the operation merely restricts already-encoded planes. The shard no longer tells the truth about which encoder produced its contents. Reconstruction and parent origin remain correct; this is an encoder provenance defect, not a claim of changed logits. It also invokes the current fixture computation if that identity has not already been memoized.

**Evidence.** `src/tessera/serving/sharding.py:660` forwards parent root rate, forests, code, superblock, and container but omits `manifest.encoder_fixture_id`. `src/tessera/unit_artifact.py:340` defaults this parameter to `_AUTO`, and line 327 resolves `_AUTO` through the current `stamped_fixture_id`. No encoder is called by the slicing operation itself. The existing readback pattern at `tests/test_slice_unit.py:270` explicitly preserves the old fixture ID.

**Reproduction.** Run the following from the repository with `PYTHONDONTWRITEBYTECODE=1 PYTHONPATH=src OMP_NUM_THREADS=1 MKL_NUM_THREADS=1 OPENBLAS_NUM_THREADS=1 python3` (CPU-only). It reads a committed fixture and does not encode new weights or write repository files.

```python
from pathlib import Path
from tessera.unit_artifact import parse_unit_artifact
from tessera.slicing import slice_unit
from tessera.serving.sharding import _reparse_shard
import tessera.encoder_identity as encoder_identity

p = parse_unit_artifact(Path(
    "tests/data/legacy/e4m3-1024-window-channel-256c.tessera"
).read_bytes())
shard = slice_unit(p, rows=(8, 16))
# Controlled stamp, to establish which identity the seam reads without
# running an encoder fixture or depending on the current encoder's value.
encoder_identity.stamped_fixture_id = lambda: b"c" * 32
q = _reparse_shard(p, shard, "rank1")
print(p.manifest.encoder_fixture_id.hex())
print(q.manifest.encoder_fixture_id.hex())
assert q.manifest.encoder_fixture_id != p.manifest.encoder_fixture_id
```

Observed parent identity: `49a718e25c916ccac1ac01c70047ed522dce59bd7fb751279846ff2313ba5182`. Observed shard identity: `6363636363636363636363636363636363636363636363636363636363636363`. The controlled in-process stub proves which stamp the seam reads without running an encoder fixture; the parent and shard are actual parsed artifacts. The equivalent defect with an untagged parent is promotion from `None` to a current encoder ID.

**Why existing tests do not negate this.** Sharding tests check reconstruction, shard geometry, and original-parent origin. They do not assert that the encoder fixture ID is unchanged through `shard_parsed_roles`/`_reparse_shard`. These checks pass because the wrong identity does not alter the restricted planes.

**Fix and acceptance.** Pass the parent's `manifest.encoder_fixture_id` explicitly when rebuilding a shard, preserving `None` as well as a noncurrent identity. Add a test where the current stamp function either returns a distinct ID or raises if called; a shard reparse must retain the parent's ID without asking the current encoder to attest old bytes. Preserve the existing correct original-parent digest/offset composition.

This is an audit finding, not a fix. The defect could be fixed locally, but the user explicitly requested a read-only code audit with issues raised and no implementation changes. No new GPU, PrismaBuild, or pytest run was performed; existing tests and served receipts were treated as evidence for the inputs they cover.

Disposition: filed for a separate implementation agent. The user explicitly requested an audit and issues, with no fixes; no implementation changes were made for this finding.
